contrib/DEVELOPMENT_WORKFLOW.md

```diff
@@ -1,63 +1,410 @@
+        - [3. Subtensor](#3-subtensor)
+    - [Notes for Commit](#notes-for-commit)
+        - [1. Commit Messages](#1-commit-messages)
+        - [2. Main branches](#2-main-branches)
+        - [3. Development Model](#3-development-model)
+        - [4. Git operations](#4-git-operations)
+    - [Notes for Pull Request](#notes-for-pull-request)
+    - [Notes for Releasing](#notes-for-releasing)
+        - [Versioning Script](#versioning-script)
+        - [Release Script](#release-script)
+        - [Security](#security)
+        - [Release Verification](#release-verification)
+        - [Post-release Actions](#post-release-actions)
+    - [Notes for Logging](#notes-for-logging)
+        - [Logging Setup](#logging-setup)
+        - [Logging Messages](#logging-messages)
+        - [Logging Variables](#logging-variables)
+        - [Logging Exceptions](#logging-exceptions)
+
+# 1. Introduction
+### What is Bittensor?
+Bittensor is an open-source project that aims to create a decentralized network for AI model training. It allows AI models to learn from each other in a decentralized manner, improving their performance and capabilities.
+
+### What is the purpose of Bittensor?
+
+The purpose of Bittensor is to democratize AI model training. By creating a decentralized network, it allows anyone to contribute to the training of AI models and benefit from their use.
+
+# 2. Architecture
+### Overview of Bittensor's architecture
+
+Bittensor's architecture consists of several key components, including the Bittensor protocol, the wallet, neurons, subtensor, and the Metagraph. These components work together to create a decentralized network for AI model training.
+
+### How the different components interact?
+
+Neurons in the Bittensor network interact with the Metagraph, a decentralized ledger that keeps track of the state of the network. The wallet is used to manage the tokens that are used for incentives in the network. The subtensor is a lower-level protocol that handles communication between neurons.
+
+# 3. Main Components
+### Bittensor Protocol
+
+The Bittensor protocol is the backbone of the network. It defines how neurons communicate with each other and with the Metagraph.
+
+### Wallet
+
+The wallet is used to manage the Bittensor tokens that are used as incentives in the network. It allows users to earn tokens for contributing to the training of AI models and spend tokens to use trained models.
+
+### Neurons
+
+Neurons are the nodes in the Bittensor network. They represent AI models that are learning from the network.
+
+### Subtensor
+
+Subtensor is a lower-level protocol that handles communication between neurons. It ensures that data is transfered securely and efficiently across the network.
+
+### Metagraph
+
+The Metagraph is a decentralized ledger that keeps track of the state of the Bittensor network. It records which neurons are part of the network and how they are connected.
+
+# 4. Preliminary Developer Notes
```

Comment:
Ditto.  Nice information, but how does this relate to development workflow?  Is likely better suited documentation material.